qlib/RestHandler.qm

```diff
@@ -614,14 +631,29 @@ public namespace RestHandler {
 
     #! the base abstract class for REST handler classes
     public class AbstractRestClass {
-        public {}
+        public {
+            #! class hash: name -> AbstractRestClass
+            hash<string, AbstractRestClass> class_hash;
+
+            #! set of REST class method names based on basic HTTP methods
+            const RestBasicMethodSet = map {$1: True}, RestHandler::Methods.iterator();
+        }
 
         constructor() {
         }
 
         #! this provides the name of the REST class
         abstract string name();
 
+        #! adds a REST class to the handler
+        addClass(AbstractRestClass cls) {
+            string name = cls.name();
+            if (class_hash{name}) {
+                throw "RETSHANDLER-ERROR", sprintf("class %y has already been registered with this handler", name);
```

Comment:
typo

Reply:
thanks - fixed - that was there for a really long time - this code was just moved to a base class